lib/MusicBrainz/Server/Report/SeriesContainingDuplicates.pm

```diff
@@ -0,0 +1,105 @@
+package MusicBrainz::Server::Report::SeriesContainingDuplicates;
+use Moose;
+
+use MusicBrainz::Server::Constants qw( %PART_OF_SERIES );
+use MusicBrainz::Server::Data::Utils qw( type_to_model );
+use MusicBrainz::Server::Entity::Util::JSON qw( to_json_object );
+
+with 'MusicBrainz::Server::Report::SeriesReport',
+     'MusicBrainz::Server::Report::FilterForEditor::SeriesID';
+
+around inflate_rows => sub {
+    my $orig = shift;
+    my $self = shift;
+
+    my $items = $self->$orig(@_);
+
+    for my $result (@$items) {
+        my $series = $result->{series};
+        my $series_type = $self->c->model('SeriesType')->get_by_id($series->{typeID});
+        my $entity_type = $series_type->item_entity_type;
+        my $model = $self->c->model(type_to_model($entity_type));
+        my $entity = $model->get_by_id($result->{entity_id});
```

Comment:
We should try to avoid calling `get_by_id` in a loop, since `get_by_ids` (plural) can easily be used to fetch everything in two queries before the loop begins.

Reply:
Oops, I was thinking the items were also sorted by entity type (so there'd generally be one type per page). But yeah, you'd still need to partition even if that were true.

It's definitely worth reducing the number of round-trips to the database. If there are 100 items per page, that's 200 round-trips to postgres or Redis * the connection latency, which isn't an effective use of resources on the Perl or pgbouncer side. That can be reduced to a maximum of 7 requests, if my math is right (one for the series type and one for each item type on the page).